Add documentation for assistant focused component triggers and custom error messages (#912)
livingdocsIO/documentation: content/customising/assistants/_index.md

{{< img src="./assistant-proposal.png" alt="K-Menu dialog with single Assistant Proposal" >}}
{{< img src="./assistant-metadata-button.png" alt="Assistant Buttons next to Metadata Form Fields" >}}

With `release-2024-11` we added the following improvements:
- Showing custom error messages

> - Assistant Butons next to a focused component

Reviewer: ```suggestion
- Assistant Buttons next to a focused component
```

      contentTypes: ['exampleContentType']
    },
    showAssistantTriggerButton: {
      metadataPropertyName: 'examplePropertyName',

>       focusedComponentName: 'exampleComponentHandle'

Reviewer: I would also add a small paragraph explaining the trigger buttons (similar to the release notes)

See example assistant above for type details.

## Error responses

A default error response is shown in cases when an assistant fails. With `release-2024-11` we added two ways to respond with a custom
error message.

> In the `assist` function either an `error` object has to be returned in the response:

Reviewer: ```suggestion
In the `assist` function either an `error` object can be returned in the response:
```

See example assistant above for type details.

## Trigger Buttons

> Trigger buttons have been added to metadata form fields as well as components. These enable users to trigger an assistant directly on that component / metadata field. To configure them the `showAssistantTriggerButton` config has to be set. 

Reviewer: I would also include the description of `metadataPropertyName` and `focusedComponentName` here (so that users don't have to go through mutliple release notes), mentioning the special behaviours, such as:

"If two 
or more assistants are registered a k-menu will be opened only with the available assistants on this focused component. Assistants which are registered
on a focused component are not shown in the normal k-menu. "
